d2layouts/d2sequence/layout.go

```diff
@@ -88,7 +87,7 @@ func layoutSequenceDiagram(g *d2graph.Graph, obj *d2graph.Object) (*sequenceDiag
 	var edges []*d2graph.Edge
 	for _, edge := range g.Edges {
 		// both Src and Dst must be inside the sequence diagram
-		if strings.HasPrefix(edge.Src.AbsID(), obj.AbsID()) && strings.HasPrefix(edge.Dst.AbsID(), obj.AbsID()) {
```

Comment:
does it work if you just change this to `obj.AbsID() + "."`?

Reply:
yeah that works but it is traversing all the way up the parent chain to the root for both nodes, while building the ID string for each of those, and then doing the string prefix comparison, which is a fairly round-about way to check if it is a descendant. I saw it more as you shouldn't need to do all of that when you can just check if it is a descendant looking up the parent chain.